drivers/pinmux/pinmux_esp32.c

```diff
@@ -180,7 +180,7 @@ static int pinmux_initialize(const struct device *device)
 /* Initialize using PRE_KERNEL_1 priority so that GPIO can use the pin
  * mux driver.
  */
-DEVICE_AND_API_INIT(pmux_dev, CONFIG_PINMUX_NAME,
-		    &pinmux_initialize, NULL, NULL,
+DEVICE_DT_INST_DEFINE(0, &pinmux_initialize,
```

Comment:
drivers/gpio/gpio_esp32.c should be updated to use `DEVICE_DT_GET()` rather than `device_get_binding(CONFIG_PINMUX_NAME)` to make sure the two are not inconsistent.

This will leave hsdk as the only pinmux infrastructure depending on the Kconfig, AFAICT.

Reply:
Thanks for the catch, was trying to grep for stuff like that, but missed it.

Reply:
This caused problems; see #30941